drivers/modem/Kconfig

```diff
@@ -30,15 +30,66 @@ config MODEM_RECEIVER
 	  in that callbacks are executed in a different work queue and data is
 	  passed around in k_pipe structures.
 
+if MODEM_RECEIVER
+
 config MODEM_RECEIVER_MAX_CONTEXTS
 	int "Maximum number of modem receiver contexts"
-	depends on MODEM_RECEIVER
 	range 1 10
 	default 1
 	help
 	  Maximum number of modem receiver contexts to handle.  For most
 	  purposes this should stay at 1.
 
+endif # MODEM_RECEIVER
```

Comment:
Is this change in case other symbols that depend on `MODEM_RECEIVER` get added later?

The `depends on` is simpler otherwise.

Reply:
Yep.  Will change back.